File: src/main/java/com/koddy/server/coffeechat/application/adapter/MeetingLinkTokenCashier.java
package com.koddy.server.coffeechat.application.adapter;

import java.time.Duration;

public interface MeetingLinkTokenCashier {
    void storeViaPlatformId(final long platformId, final String oAuthAccessToken, final Duration duration);

    String getViaPlatformId(final long platformId);

    boolean containsViaPlatformId(final long platformId);

    void storeViaMeetingId(final String meetingId, final String oAuthAccessToken, final Duration duration);

    String getViaMeetingId(final String meetingId);

    boolean containsViaMeetingId(final String meetingId);

    void deleteViaMeetingId(final String meetingId);
}

File: src/main/java/com/koddy/server/coffeechat/application/adapter/MeetingLinkTokenCashier.kt
package com.koddy.server.coffeechat.application.adapter

import java.time.Duration

interface MeetingLinkTokenCashier {
    fun storeViaPlatformId(
        platformId: Long,
        oAuthAccessToken: String,
        duration: Duration,
    )

    fun getViaPlatformId(platformId: Long): String

    fun containsViaPlatformId(platformId: Long): Boolean

    fun storeViaMeetingId(
        meetingId: String,
        oAuthAccessToken: String,
        duration: Duration,
    )

    fun getViaMeetingId(meetingId: String): String

    fun containsViaMeetingId(meetingId: String): Boolean

    fun deleteViaMeetingId(meetingId: String)
}
